### ISSUE TITLE
Multi-line comments use `#` or `#=`?

### ISSUE THREAD

**Issue by mattBrzezinski** [invenia/BlueStyle#40]
Looking at the guide I came across this example:

```julia
# Yes:

# Number of nodes to predict. Again, an issue with the workflow order. Should be updated
# after data is fetched.
p = 1

# No:

p = 1  # Number of nodes to predict. Again, an issue with the workflow order. Should be
# updated after data is fetched.
```

What are peoples thoughts on this style:
```julia
#=
Number of nodes to predict. Again, an issue with the workflow order. Should be
updated after data is fetched. 
=#
p = 1
```

**Comment by nickrobinson251** [comment]
i don't think we take a stance on that currently - do you think we should?

if we were summing up what we do in practice I think it is:
- prefer commenting each line with `#` rather than using `#= =#`
- `#= =#` can be used in the rare case you need very long comment blocks e.g. > 5 lines
- do not use `#= =#` inside code blocks, e.g. inside a function or struct

**Comment by mattBrzezinski** [comment]
I just noticed we didn't cover it in this style guide, and its bound to come up at some point.

I think your summary makes sense!

**Comment by oxinabox** [comment]
> do not use #= =# inside code blocks, e.g. inside a function or struct

Why?

I am guessing reasoning is that if a function needs that much comment it should have been multiple functions?

idk some things are hard.

**Comment by omus** [comment]
I tend to use `#= =#` only for commenting out code as doing so makes viewing diffs less noisy

**Comment by oxinabox** [comment]
Oh is that what you are doing there

**Comment by iamed2** [comment]
I don't think we should take a stance on this. TBH I think the reason we don't use block comments as much is because they arrived later, and because Python and MATLAB both don't have them.

**Comment by JeffFessler** [comment]
Actually Matlab does have block comments:
https://www.mathworks.com/help/matlab/matlab_prog/comments.html
They are quite useful both for commenting out code blocks and for making "longer" comments more readable.  But "longer" should be left to the eye of the beholder (or coder).

**Comment by iamed2** [comment]
> Actually Matlab does have block comments

😮
